src/main/java/org/jabref/gui/entryeditor/LatexCitationsTabViewModel.java

```diff
@@ -160,19 +170,41 @@ private List<Path> searchDirectory(Path directory) {
         }
     }
 
+    private void setListener(Path path) {
+        if (!hasListener) {
+            try {
+                fileUpdateMonitor.addListenerForFile(path, this::refreshLatexDirectory);
+                hasListener = true;
+            } catch (IOException e) {
+                LOGGER.error("Could not find file", e);
+            }
+        }
+    }
+
+    private void setListener(Path oldPath, Path newPath) {
+        try {
+            fileUpdateMonitor.removeListener(oldPath, this::refreshLatexDirectory);
+            fileUpdateMonitor.addListenerForFile(newPath, this::refreshLatexDirectory);
+        } catch (IOException e) {
+            LOGGER.error("Could not find file", e);
+        }
+    }
+
     public void setLatexDirectory() {
         DirectoryDialogConfiguration directoryDialogConfiguration = new DirectoryDialogConfiguration.Builder()
                 .withInitialDirectory(directory.get()).build();
 
-        dialogService.showDirectorySelectionDialog(directoryDialogConfiguration).ifPresent(selectedDirectory ->
-                databaseContext.getMetaData().setLatexFileDirectory(preferencesService.getFilePreferences().getUserAndHost(), selectedDirectory.toAbsolutePath()));
+        dialogService.showDirectorySelectionDialog(directoryDialogConfiguration).ifPresent(selectedDirectory -> {
+                    databaseContext.getMetaData().setLatexFileDirectory(preferencesService.getFilePreferences().getUserAndHost(), selectedDirectory.toAbsolutePath());
+                    setListener(directory.get(), selectedDirectory);
+        });
 
         init(currentEntry);
     }
 
     public void refreshLatexDirectory() {
         latexParserResult = null;
-        init(currentEntry);
+        Platform.runLater(() -> init(currentEntry));
```

Comment:
At most one scan job should wait. Meaning: If the changes on disk are incoming in a faster rate than the interpretation, it could lead to issues. Maybe not a real issue... Needs to be tried out.